sqlmesh/core/dialect.py

```diff
@@ -839,3 +839,13 @@ def transform_values(
             yield exp.func("PARSE_JSON", f"'{value}'")
         else:
             yield value
+
+
+def to_schema(sql_path: str | exp.Table) -> exp.Table:
+    if isinstance(sql_path, exp.Table):
```

Comment:
you should check if table.this is filled, if it is, you need to swap stuff around, this ensures passing in a table will convert it into a schema